frontend/ui/widget/filechooser.lua

```diff
@@ -112,33 +244,29 @@ function FileChooser:getList(path, collate)
         unreadable_dir_content[path] = nil
         for f in iter, dir_obj do
             if self.show_hidden or not util.stringStartsWith(f, ".") then
-                local filename = path.."/"..f
-                local attributes = lfs.attributes(filename)
-                if attributes ~= nil then
-                    local item = true
-                    if attributes.mode == "directory" and f ~= "." and f ~= ".." then
-                        if self:show_dir(f) then
-                            if collate then -- when collate == nil count only to display in folder mandatory
-                                item = FileChooser.getListItem(f, filename, attributes)
-                            end
-                            table.insert(dirs, item)
-                        end
-                    -- Always ignore macOS resource forks.
-                    elseif attributes.mode == "file" and not util.stringStartsWith(f, "._") then
-                        if self:show_file(f, filename) then
-                            if collate then -- when collate == nil count only to display in folder mandatory
-                                item = FileChooser.getListItem(f, filename, attributes, collate)
-                            end
-                            table.insert(files, item)
-                        end
+                local fullpath = path.."/"..f
+                local attributes = lfs.attributes(fullpath) or {}
+                local item = true
+                if attributes.mode == "directory" and f ~= "." and f ~= ".."
+                        and self:show_dir(f) then
+                    if collate then -- when collate == nil count only to display in folder mandatory
+                        item = self:getListItem(path, f, fullpath, attributes, collate)
+                    end
+                    table.insert(dirs, item)
+                -- Always ignore macOS resource forks.
+                elseif attributes.mode == "file" and not util.stringStartsWith(f, "._")
+                        and self:show_file(f, fullpath) then
+                    if collate then -- when collate == nil count only to display in folder mandatory
+                        item = self:getListItem(path, f, fullpath, attributes, collate)
                     end
+                    table.insert(files, item)
                 end
             end
         end
     else -- error, probably "permission denied"
         if unreadable_dir_content[path] then
             -- Add this dummy item that will be replaced with a message by genItemTable()
-            table.insert(dirs, FileChooser.getListItem("./.", path, lfs.attributes(path)))
+            table.insert(dirs, FileChooser:getListItem(path, "./.", path, lfs.attributes(path)), collate)
```

Comment:
(Trying to update my "browse by metadata POC" patch,) and noticed this has gone previously from  `FileChooser:getListItem()` (method) to  `FileChooser.getListItem()` (module function, #10994) and back to `FileChooser:getListItem()` with this PR.
Is that ok with you @hius07 ?
Anyway, there's this one that jumps out as being wrongly/inconsistently `FileChooser:getListItem()`, and it probably should be `self:getListItem()` ?

Reply:
For context, `FileChooser:getListItem` calls `FileChooser:getMenuItemMandatory`, which may call `FileChooser:getList` for getting directory contents, which is why it reverted to `:`. However, there doesn't seem to be a good reason for `FileChooser:getList` to not be `FileChooser.getList`, and if we change that (including `show_file` and `show_dir`), we can make all functions involved here `.`.

> Anyway, there's this one that jumps out as being wrongly/inconsistently FileChooser:getListItem(), and it probably should be self:getListItem() ?

You're correct here.

Reply:
> there doesn't seem to be a good reason

https://github.com/koreader/koreader/blob/aabd6d7a2602e75d7e12d4d8e48f291f7e11275e/frontend/ui/widget/filechooser.lua#L304